spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -681,68 +712,68 @@ def time_to_frame(self, time):
-        if hasattr(sorting, '_t0'):
-            t0 = sorting._t0
+        
+        all_properties = set()
+        all_features = set()
+        for unit_id in unit_ids:
+            all_properties.update(sorting.get_unit_property_names(unit_id))
+            all_features.update(sorting.get_unit_spike_feature_names(unit_id))
+        
+        if property_descriptions is not None:
+            property_description_names = set(property_descriptions.keys())
+            print(all_properties)
+            print(property_description_names)
+            if not all_properties == property_description_names:
+                print('Warning: property_descriptions were specified, ' +
+                      'but not all unit properties have a description!')
+                for pr in all_properties - property_description_names:
+                    property_descriptions.update({pr: 'no description'})
         else:
-            t0 = 0.
-
-        (all_properties, all_features) = find_all_unit_property_names(
-            properties_dict=sorting._properties,
-            features_dict=sorting._features
-        )
-
-        if os.path.exists(save_path):
-            read_mode = 'r+'
-        else:
-            read_mode = 'w'
-
-        with NWBHDF5IO(save_path, mode=read_mode) as io:
-            if read_mode == 'r+':
-                nwbfile = io.read()
-            else:
-                kwargs = {'session_description': 'No description',
-                          'identifier': str(uuid.uuid4()),
-                          'session_start_time': datetime.now()}
-                kwargs.update(**nwbfile_kwargs)
-                nwbfile = NWBFile(**kwargs)
-
-            # If no Units present in nwb file
-            if nwbfile.units is None:
-                for id in ids:
-                    spkt = sorting.get_unit_spike_train(unit_id=id) / fs # spike trains withinin the SortingExtractor object are not scaled by sampling frequency
-                    nwbfile.add_unit(id=id, spike_times=spkt)
-
-            # Units properties
+            property_descriptions = {}
             for pr in all_properties:
-                unit_ids = [int(k) for k, v in sorting._properties.items()
-                            if pr in v]
-                vals = [v[pr] for k, v in sorting._properties.items()
-                        if pr in v]
-                set_dynamic_table_property(
-                    dynamic_table=nwbfile.units,
-                    row_ids=unit_ids,
-                    property_name=pr,
-                    values=vals,
-                    default_value=np.nan,
-                    description='no description'
-                )
-
+                property_descriptions.update({pr: 'no description'})
+        
+        # If no Units present in nwb file
+        if nwbfile.units is None:
+            for pr in all_properties:
+                # Special case of setting max_electrodes requires a table to be
```

Comment:
```suggestion
                desc = property_descriptions.get(pr, 'no description')
                # Special case of setting max_electrodes requires a table to be
```

This should obviate the dictionary building above